### DISCUSSION TITLE
Section 4 Exercise/ Ethernaut Level 9 (King): transfer() and send() fails in attaining Kingship, but call() succeeds

### DISCUSSION THREAD

**Discussion by RichuAK** [Cyfrin/security-and-auditing-full-course-s23#84] [Q&A] [answered]
I was trying to solve the [King](https://ethernaut.openzeppelin.com/level/9) challenge in Ethernaut, but taking Kingship proved to be a bit tricky:

```solidity

    // Transaction fails
    function takeTheThroneWithTransfer() public payable{
        payable(kingContractAddress).transfer(msg.value);
    }

    // Transaction fails
    function takeTheThroneWithSend() public payable {
        bool success = payable(kingContractAddress).send(msg.value);
        require(success, "Failed to send Ether");
    }

    // Transaction succeeds, attains Kingship
    function takeTheThroneWithCall() public payable{
        (bool success, ) = kingContractAddress.call{value: msg.value}("");
        require(success, "Failed to send Ether");
    }
```
I want to understand why this is the case, why transfer() and send() doesn't work but call() does. 

Once Kingship was attained, I was able to submit the level successfully, solving the challenge. 

I asked the [same question](https://ethereum.stackexchange.com/questions/158975/ethernaut-king-transfer-and-send-doesnt-work-but-call-succeeds-in-takin) on stackexchange as well.

Thanks in advance!

**Comment by johnsonstephan** [comment, marked as answer]
Good walkthrough here to help understand why `transfer()` and `send()` don't work but `call()` does - https://dev.to/nvn/ethernaut-hacks-level-9-king-12

**Reply by RichuAK** [reply]
Thank you so much! This was exactly what I was looking for! Couldn't find it myself after a lot of search!

**Reply by RichuAK** [reply]
@johnsonstephan 

Feel free to answer the [stackexchange question](https://ethereum.stackexchange.com/questions/158975/ethernaut-king-transfer-and-send-doesnt-work-but-call-succeeds-in-takin) as well, or I'll answer it by tomorrow. Thanks a ton again!

**Reply by johnsonstephan** [reply]
I'm glad to hear it helped! I've answered on Stack as well -- https://ethereum.stackexchange.com/a/158985/130042
